xtask/src/main.rs

```diff
@@ -323,6 +344,66 @@ fn run_fmt_project(fmt_opt: &FmtOpt) -> Result<()> {
     Ok(())
 }
 
+/// Check that SPDX file headers are present.
+///
+/// If not in check mode, add any missing headers.
+fn format_file_headers(fmt_opt: &FmtOpt) -> Result<()> {
+    // Recursively get Rust files
+    let mut cmd = Command::new("git");
+    cmd.args(["ls-files", "*.rs"]);
+    let output = cmd.output()?;
+    if !output.status.success() {
+        bail!("command failed: {}", output.status);
+    }
+    let paths = std::str::from_utf8(&output.stdout)?.lines();
+
+    // Path prefixes that should not be checked/formatted.
+    let exclude_path_prefixes = [
+        // A user copying the template or uefi-std-example does not need to use
+        // our license.
+        "template/src/main.rs",
+        "uefi-std-example/src/main.rs",
+        // This directory contains short code snippets used in `trybuild` tests,
+        // no license needed.
+        "uefi-macros/tests/ui/",
+    ];
+
+    let expected_header = "// SPDX-License-Identifier: MIT OR Apache-2.0\n\n";
```

Comment:
maybe also make this a constant; close to the new constant mentioned above?

Reply:
done